browser/src/control/Control.JSDialogBuilder.js

```diff
@@ -2949,7 +2949,10 @@ L.Control.JSDialogBuilder = L.Control.extend({
 
 		switch (data.action_type) {
 		case 'grab_focus':
-			control.focus();
+			// We will ignore focus on a specific element of the widget if the widget is tabControl.
+			// Focus should be on Tab and not on any element inside dialog if tabcontrols are present
+			if (!document.getElementById('tabcontrol'))
```

Comment:
Please do not use hardcoded ids, also this looks for any element in the whole webpage, not in the current dialog's container.

Reply:
Yes i think i can improve this here to only search for `tabcontrol` for open dialogs and not in entire document ?

Reply:
@eszkadev in this type of focus function or action handler function does not have proper widget type information. So what can be other way to see if the opened jsDialog on screen have the tab controls ?

I understand the use of hard-coded id is not ok but to fix this focus issue we need to make some exception here maybe i am wrong please guide me here

Reply:
I think we can manage to do better solution - not prevent `grab_focus` anytime we have `tabcontrol` element in the whole document.

eg. better would be to check `control` 's node classes, if they contain class which is set for the root container in the tabcontrol widget.

Reply:
>eg. better would be to check control 's node classes, if they contain class which is set for the root container in the tabcontrol widget.

@eszkadev i did not get this point 🤔

Reply:
`control.classList.contains('ui-some-widget-class')`